server/routerlicious/packages/lambdas/src/scribe/lambda.ts

```diff
@@ -759,6 +759,14 @@ export class ScribeLambda implements IPartitionLambda {
 			this.validParentSummaries = [];
 		}
 		this.validParentSummaries.push(summaryHandle);
+		const countOverLimit =
+			this.validParentSummaries.length -
+			this.serviceConfiguration.scribe.maxTrackedServiceSummaryVersionsSinceLastClientSummary;
+		if (countOverLimit === 1) {
+			this.validParentSummaries.shift();
+		} else if (countOverLimit > 1) {
+			this.validParentSummaries.splice(0, countOverLimit);
+		}
```

Comment:
why not just do:
```
if (countOverLimit>0){
this.validParentSummaries.splice(0, countOverLimit);
}
```

Reply:
[Shift() is over 2x more performant than splice()](https://www.measurethat.net/Benchmarks/Show/12324/0/slice-vs-splice-vs-shift), so if we can use `shift()` that will be better. Splice() should just be used for cases where a document was too large before this change.

Reply:
Maybe consider to add this comment into the `if elseif` part? I personally think we don't have to do the split but if e want to do, maybe add a comment here just state for performance purpose so people won't get confused like me?